server/src/test/java/org/elasticsearch/snapshots/DedicatedClusterSnapshotRestoreIT.java

```diff
@@ -981,6 +987,38 @@ public void testRestoreShrinkIndex() throws Exception {
         ensureYellow();
     }
 
+    public void testSnapshotWithDateMath() {
+        final String repo = "repo";
+        final AdminClient admin = client().admin();
+
+        final IndexNameExpressionResolver nameExpressionResolver = new IndexNameExpressionResolver(Settings.EMPTY);
+        final String snapshotName = "<snapshot-{now/d}>";
+
+        logger.info("-->  creating repository");
+        assertAcked(admin.cluster().preparePutRepository(repo).setType("fs")
+            .setSettings(Settings.builder().put("location", randomRepoPath())
+                .put("compress", randomBoolean())));
+
+        final String expression1 = nameExpressionResolver.resolveDateMathExpression(snapshotName);
+        logger.info("-->  creating date math snapshot");
+        CreateSnapshotResponse snapshotResponse =
+            admin.cluster().prepareCreateSnapshot(repo, snapshotName)
+                .setIncludeGlobalState(true)
+                .setWaitForCompletion(true)
+                .get();
+        assertThat(snapshotResponse.status(), equalTo(RestStatus.OK));
+        // snapshot could be taken before or after a day rollover
+        final String expression2 = nameExpressionResolver.resolveDateMathExpression(snapshotName);
+
+        SnapshotsStatusResponse response = admin.cluster().prepareSnapshotStatus(repo)
+            .setSnapshots(Sets.newHashSet(expression1, expression2).toArray(Strings.EMPTY_ARRAY))
```

Comment:
this is a varargs method, so just `.setSnapshots(expression1, expression2)` is good enough.

Reply:
w/o a day rollover it requests two statuses with two identical names - and the result is 2 snapshot statuses - that's why I used **set** here

Reply:
ok